### ISSUE TITLE
Encountered an error when trying to set a custom url

### ISSUE THREAD

**Issue by maxloh** [spikespaz/search-deflector#12]
I have encountered an error when I tried to set search engine to "google.com.hk/search?q={{query}}"

----------------------------------------------------

The error information follows below.

**File:** `std\encoding.d`

**Line:** 2554

**Message:**
```
Unrecognized Encoding: big5
```

**Stack Trace:**
```
0x00007FF7632AF0CD
0x00007FF7632AEE26
0x00007FF7632A7CDC
0x00007FF76329477C
0x00007FF763252F4D
0x00007FF76323300A
0x00007FF763233610
0x00007FF763232C97
0x00007FF7632311E0
0x00007FF763236635
0x00007FF763235C8E
0x00007FF763235AD0
0x00007FF76329AD8D
0x00007FF76329AC3A
0x00007FF7632B9658
0x00007FFE8A7B3704 in BaseThreadInitThunk
0x00007FFE8C4DA1D1 in RtlUserThreadStart

```

**Comment by spikespaz** [comment]
Honestly I'm clueless, I don't know why this would have happened.

**Comment by spikespaz** [comment]
Is this during setup or is this after using the custom URL and making a search?

**Comment by maxloh** [comment]
I encounter the error when I setup search deflector using the configure shortcut in start menu
By typing URL, I am able to set custom URL to `google.ca/search?q={{query}}` 

![screenshot 1](https://user-images.githubusercontent.com/35831069/44723875-c9946f80-ab03-11e8-907c-ed3db2632ff2.PNG)

but failed to set it to `google.com.hk/search?q={{query}}`

![screenshot 2](https://user-images.githubusercontent.com/35831069/44723876-c9946f80-ab03-11e8-977f-a4b334927cbd.PNG)
--------------
The only solution I found is editing `EngineURL` varieable in `HKEY_CURRENT_USER\Software\Clients\SearchDeflector`
After that, it works like a charm.

Maybe you can include the registry way of setup in the tutorial.

**Comment by spikespaz** [comment]
To be honest this is incredibly strange. My code isn't handling anything improperly as far as I can tell. I think it's an issue with the Phobos standard library.

The offending lines are here: https://github.com/spikespaz/search-deflector/blob/master/source/setup.d#L253-L278

But I can't figure hot which function is causing the error in `std.encoding` because the stack trace isn't giving me debug information like it should be. I must have done something wrong when making the error reporting section too.

**Comment by maxloh** [comment]
When the error occur, an `EncodingException` is thrown, see https://github.com/dlang/phobos/blob/master/std/encoding.d#L2644-L2645

I think this is a bug of the get() function in line 286 of `setup.d` because someone faced this error and their code contains the get() function too, see https://forum.dlang.org/post/oo42dq$2ur3$1@digitalmars.com

Besides the CurlException, you may also throw `EncodingException` in line 285 to 292 of `setup.d` and skip the validate step by always returning true 
Maybe it could fix this bug

**Comment by spikespaz** [comment]
Finally resolved this in 53fe86f. `std.net.curl` doesn't support your encoding, but fetching the site data with a get request wasn't necessary anyway and was wasteful. I replaced the relevant section in `validateEngineUrl` with an address resolution through `std.socket`. This should work better than the original implementation anyway.
